### PR TITLE
Improve robots.txt generation

### PR DESCRIPTION
**Pull request by Hlavtox**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 9.0.x
| Description?      | Improves robots.txt generation. See below
| Type?             | refacto
| Category?         | FO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | Compare before & after in https://www.diffchecker.com
| UI Tests          | https://github.com/florine2623/testing_pr/actions/runs/12065224542 ✅ 
| Fixed issue or discussion?     | Fixes https://github.com/PrestaShop/PrestaShop/issues/36991
| Related PRs       | 
| Sponsor company   | TRENDO s.r.o.

### Description
- Refactors generation of robots.txt
- Adds comments and better explanation
- Fixes issue when one empty rewrite could break whole indexation of the site
- Adds rules for non-prefixed default language to account for https://github.com/PrestaShop/PrestaShop/pull/37236/

### Before
[robots.txt](https://github.com/user-attachments/files/17933985/robots.txt)

### After
[robots.txt](https://github.com/user-attachments/files/17933972/robots.txt)

### CODE REVIEW

File: classes/Tools.php
Comment (Touxten) [inline classes/Tools.php:2561]: :rofl:

**Review by Touxten — APPROVED** [review]

**Review by ShaiMagal — APPROVED** [review]
LGTM

**Review by kpodemski — APPROVED** [review]

**Review by florine2623 — APPROVED** [review]
Hello @Hlavtox ,

Tested with and without PR !

Thanks for share the diffchecker site ^^it helped a lot !

The robots.txt file is more readable than before ! Titles on the sitemap are updated with the new ones !

It is QA ✅

**Comment by ps-jarvis** [comment]
PR merged, well done!

Message to @PrestaShop/committers: do not forget to milestone it before the merge.

**Pull request merged**
